src/jenkins.py

```diff
@@ -144,6 +146,467 @@ def _get_api_credentials(container: ops.Container) -> Credentials:
         raise JenkinsBootstrapError("Admin API credentials not yet setup.") from exc
 
 
+class Jenkins:
+    """Wrapper for Jenkins functionality.
+
+    Attrs:
+        environment: the Jenkins environment configuration.
+        web_url: the Jenkins web URL.
+        login_url: the Jenkins login URL.
+        version: the Jenkins version.
+    """
+
+    environment: Environment
+
+    def __init__(self, environment: Environment):
+        """Construct a Jenkins class.
+
+        Args:
+            environment: the Jenkins environment.
+        """
+        self.environment = environment
+
+    @property
+    def web_url(self) -> str:
+        """Get the Jenkins web URL.
+
+        Returns: the web URL.
+        """
+        env_dict = typing.cast(typing.Dict[str, str], self.environment)
+        return f"http://localhost:{WEB_PORT}{env_dict['JENKINS_PREFIX']}"
+
+    @property
+    def login_url(self) -> str:
+        """Get the Jenkins login URL.
+
+        Returns: the login URL.
+        """
+        return f"{self.web_url}{LOGIN_PATH}"
+
+    @property
+    def version(self) -> str:
```

Comment:
Since we only expect the server version to change during a charm upgrade, should we just use a constant here?

Reply:
That would mean having the version defined in two places, the rockfile and the charm code. Anyway, this was already the existing behaviour, so I'd rather keep it as it is at least in this PR